simbad/command_line/simbad_database.py

```diff
@@ -163,6 +189,151 @@ def create_lattice_db(database):
+        raise RuntimeError(msg)
+
+    dimple.contaminants.prepare.main(verbose=False)
+
+    simbad_dat_path = os.path.join(database, '*', '*', '*', '*.dat')
+    existing_dat_files = [os.path.basename(f).split('.')[0] for f in glob.glob(simbad_dat_path)]
+    erroneous_files = ['4V43']
+    dimple_files = ['cached', 'data.json', 'data.py']
+
+    with open("data.json") as data_file:
+        data = json.load(data_file)
+
+    results = []
+    for child in data["children"]:
+        try:
+            for child_2 in child["children"]:
+                space_group = child_2["name"].replace(" ", "")
+                for child_3 in child_2["children"]:
+                    pdb_code = child_3["name"].split()[0]
+                    if pdb_code in existing_dat_files or pdb_code in erroneous_files:
+                        continue
+                    uniprot_name = child["name"]
+                    uniprot_mnemonic = uniprot_name.split('_')[1]
+                    score = ContaminantSearchResult(pdb_code, space_group, uniprot_name, uniprot_mnemonic)
+                    results.append(score)
+        except KeyError:
+            pass
+
+    if len(results) == 0:
+        logger.info("Contaminant database up to date")
+    else:
+        logger.info(
+            "%d new entries were found in the contaminant database, "
+            + "updating SIMBAD database", len(results)
+        )
+
+        if "MRD_DB" in os.environ:
+            morda_installed_through_ccp4 = True
+        else:
+            morda_installed_through_ccp4 = False
+
+        if morda_installed_through_ccp4:
+            morda_dat_path = os.path.join(os.environ['MRD_DB'], 'home', 'ca_DOM', '*.dat')
+            morda_dat_files = set([os.path.basename(f) for f in glob.glob(morda_dat_path)])
+            exe = os.path.join(os.environ['MRD_PROG'], "get_model")
+        else:
+            logger.info("Morda not install locally, therefore morda domains will not be added to contaminant database")
+
+        what_to_do = []
+        for result in results:
+            stem = os.path.join(os.getcwd(), database, result.uniprot_mnemonic, result.uniprot_name,
+                                result.space_group)
+            if not os.path.exists(stem):
+                os.makedirs(stem)
+            try:
+                content = iotbx.pdb.fetch.fetch(result.pdb_code, data_type='pdb', format='pdb', mirror='pdbe')
+                download_state = content.msg
+            except RuntimeError:
+                download_state = "FAIL"
+
+            if download_state == "OK":
+                dat_content = simbad.db._to_dat(content)
+                with open(os.path.join(stem, result.pdb_code + ".dat"), "w") as f_out:
+                    f_out.write(dat_content)
+
+                if simbad.db.is_valid_dat(os.path.join(stem, result.pdb_code + ".dat")):
+                    pass
+                else:
+                    logger.debug("Unable to convert %s to dat file", result.pdb_code)
```

Comment:
This block of code seems to be a copy from one of the bits in the lattice search. If you copy-pasted this, please refactor out and call from both places the same code!